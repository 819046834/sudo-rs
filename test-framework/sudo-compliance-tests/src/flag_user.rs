use pretty_assertions::assert_eq;
use sudo_test::{As, EnvBuilder};

use crate::{Result, SUDOERS_FERRIS_ALL_NOPASSWD, SUDOERS_ROOT_ALL};

#[ignore]
#[test]
fn root_can_become_another_user() -> Result<()> {
    let env = EnvBuilder::default()
        .user("ferris", &[])
        .sudoers(SUDOERS_ROOT_ALL)
        .build()?;

    let expected = env.stdout(&["id"], As::User { name: "ferris" }, None)?;
    let actual = env.stdout(&["sudo", "-u", "ferris", "id"], As::Root, None)?;

    assert_eq!(expected, actual);

    Ok(())
}

#[ignore]
#[test]
fn user_can_become_another_user() -> Result<()> {
    let env = EnvBuilder::default()
        .user("ferris", &[])
        .user("someone_else", &[])
        .sudoers(SUDOERS_FERRIS_ALL_NOPASSWD)
        .build()?;

    let expected = env.stdout(
        &["id"],
        As::User {
            name: "someone_else",
        },
        None,
    )?;
    let actual = env.stdout(
        &["sudo", "-u", "someone_else", "id"],
        As::User { name: "ferris" },
        None,
    )?;

    assert_eq!(expected, actual);

    Ok(())
}
