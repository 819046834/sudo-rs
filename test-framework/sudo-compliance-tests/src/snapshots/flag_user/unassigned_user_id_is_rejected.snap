---
source: sudo-compliance-tests/src/flag_user.rs
expression: output.stderr()
---
sudo: unknown user: #1234
